js/ai/src/generate.ts

```diff
@@ -104,6 +105,16 @@ export class Message<T = unknown> implements MessageData {
     return this.content.find((part) => part.data)?.data as T | null;
   }
 
+  /**
+   * Returns all tool request found in this message.
+   * @returns Array of all tool request found in this message.
+   */
+  toolRequests(): ToolRequestPart[] {
+    return this.content.filter(
+      (part) => !!part.toolRequest
+    ) as ToolRequestPart[];
+  }
```

Comment:
Should this actually return the tool request itself instead of just the part? i.e. if I have: `[{text: "foo", {toolRequest: {foo: "bar"}]` should this return `[{toolRequest: {foo: "bar"}]` or should it return `[{foo: "bar"}]`?

Oh wait, parts can have metadata, nevermind.

Reply:
wait, what metadata? I think your suggestion actually makes sense...

Reply:
Done.

Reply:
oh, EmptyPartSchema has the metadata.... so yeah, metadata is at the part level... ugh... undo!